src/overlays/actors/ovl_Dm_Tag/z_dm_tag.c

```diff
@@ -31,30 +30,240 @@ const ActorInit Dm_Tag_InitVars = {
+            gSaveContext.nextTransition = 6;
+            this->unk_1A4++;
+            break;
+    }
+    return ret;
+}
+
+s32 func_80C227E8(DmTag* this, PlayState* play) {
+    if (this->unk_1A4 == 0) {
+        func_800B7298(play, &this->actor, 7);
+        play->nextEntranceIndex = 0xBC40;
+        gSaveContext.nextCutsceneIndex = 0;
+        play->sceneLoadFlag = 0x14;
+        play->unk_1887F = 2;
+        gSaveContext.nextTransition = 6;
+        this->unk_1A4++;
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tag/func_80C22350.s")
+s32* func_80C22880(DmTag* this, PlayState* play) {
+    s32 time;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tag/func_80C22400.s")
+    switch (this->unk_18E) {
+        case 1:
+            time = gSaveContext.save.time - 0x3FFC;
+            if ((time >= 0xA54B) && (time < 0xB54A) && (gSaveContext.save.day == 2)) {
+                this->unk_1A8 = func_80C227E8;
+                return D_80C22BF0;
+            }
+            return D_80C22C30;
+        case 2:
+            this->unk_1A8 = func_80C224D8;
+            return D_80C22BFC;
+    }
+    return NULL;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tag/func_80C2247C.s")
+s32 func_80C2291C(DmTag* this, PlayState* play) {
+    s32 ret = false;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tag/func_80C224D8.s")
+    if (this->unk_18C & 7) {
+        if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+            this->unk_18C |= 8;
+            SubS_UpdateFlags(&this->unk_18C, 0, 7);
+            this->unk_190 = func_80C22880(this, play);
+            this->actionFunc = func_80C229FC;
+            ret = true;
+        }
+    }
+    return ret;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tag/func_80C227E8.s")
+void func_80C229AC(DmTag* this, PlayState* play) {
+    SubS_UpdateFlags(&this->unk_18C, 3, 7);
+    this->actor.flags |= ACTOR_FLAG_1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tag/func_80C22880.s")
+void func_80C229EC(DmTag* this, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tag/func_80C2291C.s")
+void func_80C229FC(DmTag* this, PlayState* play) {
+    if (func_8010BF58(&this->actor, play, this->unk_190, this->unk_1A8, &this->unk_194)) {
+        this->actionFunc = func_80C229AC;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tag/func_80C229AC.s")
+void DmTag_Init(Actor* thisx, PlayState* play) {
+    DmTag* this = (DmTag*)thisx;
+    Player* player = GET_PLAYER(play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tag/func_80C229EC.s")
+    if (gSaveContext.save.weekEventReg[85] & 4) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tag/func_80C229FC.s")
+    if ((play->sceneNum == SCENE_YADOYA) && (play->curSpawn == 4)) {
+        player->stateFlags1 |= 0x20;
```

Comment:
```suggestion
        player->stateFlags1 |= PLAYER_STATE1_20;
```
I think you have a thing that updates enums; might want to add the Player State stuff.